bindings/python/src/transform.rs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{error, fmt};
+use iceberg::spec::Transform;
+use iceberg::transform::create_transform_function;
+use iceberg::Error;
+
+use arrow::{
+    array::{make_array, Array, ArrayData},
+};
+use arrow::pyarrow::{FromPyArrow, ToPyArrow};
+use pyo3::{exceptions::PyValueError, prelude::*};
+
+#[derive(Debug)]
+enum PyO3IcebergError {
```

Comment:
How about implementing a simple `fn to_py_err(err: iceberg::Error) -> PyError`. So we can use:

```rust
let bucket = create_transform_function(&Transform::Bucket(num_buckets)).map_err(to_py_err)?;
```

But no need for implementing `Display`, `From` by hand.

Reply:
That worked like a charm! Thanks @Xuanwo